### PR TITLE
CIRC-7740 If user_error does not exist in error response then determine error and display

### PR DESCRIPTION
**Pull request by Brady-Woods**
fixed checking for user_error when message may be undefined.
https://circonus.atlassian.net/browse/CIRC-7740

### CODE REVIEW

File: src/datasource.ts [hunk 1]
```diff
@@ -816,7 +816,7 @@ export default class IrondbDatasource extends DataSourceApi<IrondbQueryInterface
                     } catch (error) {
                         log(() => 'throwerr() failed to parse json from error.data.message. error: ' + error);
                     }
-                    if (message.user_error) {
+                    if (message && message.user_error) {
                         throw message.user_error.message + ', in query: ' + message.arguments.q;
                     } else if (error.statusText === 'Not Found') {
```
Comment (bdunavant) [inline src/datasource.ts:821]: These fallbacks to error.statusText only happen inside of the if(error.data.message) block.   This looks wrong to me.   Shouldnt' those fallbacks back outside so they can be used if there is no error.data.message?

Comment (Brady-Woods) [reply src/datasource.ts]: Commit below should address these concerns.

File: src/datasource.ts [hunk 2]
```diff
@@ -807,28 +807,34 @@ export default class IrondbDatasource extends DataSourceApi<IrondbQueryInterface
...
-            throw error ? error.toString() : 'unknown';
+            else {
+                throw JSON.stringify(error)
+            }
+        } else if (typeof error === 'string'){
+            throw JSON.stringify(error)
+        }
+        else {
+            throw JSON.stringify(error)
         }
```
Comment (bdunavant) [inline src/datasource.ts:838]: This is missing a few things.   I think this is what you are looking for (please check it though).
```
        if (typeof error === 'object') {
            if (error.data && error.data.message) {
                var message
                try {
                    message = JSON.parse(error.data.message);
                } catch {
                    throw String(error.status + ': ' + error.statusText)
                }
                if (typeof message === 'object' && message.user_error) {
                    if (message.arguments && message.arguments.q && message.user_error.message) {
                        throw message.user_error.message + ', in query: ' + message.arguments.q;
                    } else {
                        throw message.user_error
                    }
                }
                else {
                    throw JSON.stringify(message)
                }
            }
            else {
                throw JSON.stringify(error)
            }
        } 
        else {
            throw JSON.stringify(error)
        }
```

**Comment by bdunavant** [comment]
This was handled inside of https://github.com/circonus-labs/circonus-irondb-datasource/pull/80

**Pull request closed without merging**